app/src/main/kotlin/com/softteco/template/ui/components/AppPasswordField.kt

```diff
@@ -20,42 +22,54 @@ import androidx.compose.material3.OutlinedTextField
 import androidx.compose.material3.OutlinedTextFieldDefaults
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.mutableStateOf
 import androidx.compose.runtime.remember
 import androidx.compose.runtime.setValue
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.platform.SoftwareKeyboardController
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.input.ImeAction
 import androidx.compose.ui.text.input.KeyboardType
 import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.unit.dp
 import com.softteco.template.R
-import com.softteco.template.ui.feature.PasswordFieldState
 import com.softteco.template.ui.theme.Dimens
 
 @OptIn(ExperimentalComposeUiApi::class)
 @Composable
 fun PasswordField(
     passwordValue: String,
     onPasswordChanged: (String) -> Unit,
-    fieldStatePassword: PasswordFieldState,
+    fieldStatePassword: TextFieldState,
+    onInputComplete: () -> Unit,
     modifier: Modifier = Modifier,
 ) {
+    var passwordVisibility by remember { mutableStateOf(true) }
+    val isError = fieldStatePassword is TextFieldState.Error
+    val keyboardController: SoftwareKeyboardController? = LocalSoftwareKeyboardController.current
+    var isFocused by remember { mutableStateOf(false) }
+
+    val isKeyboardVisible = WindowInsets.ime.getBottom(LocalDensity.current) > 0
+    LaunchedEffect(isKeyboardVisible) { if (!isKeyboardVisible) onInputComplete() }
```

Comment:
There are 2 issues here:
1. onInputComplete() triggers on the login screen when the user has not entered anything
2. onInputComplete() does not work when opening/closing the keyboard